src/validation/schema.js

```diff
@@ -17,21 +17,33 @@ const BUILTIN_SCALAR_TYPES = [
 ]
 
 // Type suggestions for common mistakes
-const TYPE_SUGGESTIONS = {
-  Address: 'Bytes',
-  address: 'Bytes',
-  bytes: 'Bytes',
-  string: 'String',
-  bool: 'Boolean',
-  boolean: 'Boolean',
-  Bool: 'Boolean',
-  int: 'Int',
-  float: 'Float',
-  uint128: 'BigInt',
-  int128: 'BigInt',
-  uint256: 'BigInt',
-  int256: 'BigInt',
-}
+const TYPE_SUGGESTIONS = [
+  ['Address', 'Bytes'],
+  ['address', 'Bytes'],
+  ['bytes', 'Bytes'],
+  ['string', 'String'],
+  ['bool', 'Boolean'],
+  ['boolean', 'Boolean'],
+  ['Bool', 'Boolean'],
+  ['float', 'BigDecimal'],
+  ['Float', 'BigDecimal'],
+  ['int', 'Int'],
+  ['uint', 'BigInt'],
+  [/^(u|uint)(8|16|24)$/, 'Int'],
+  [/^(i|int)(8|16|24|32)$/, 'Int'],
+  [/^(u|uint)32$/, 'BigInt'],
+  [
+    /^(u|uint|i|int)(40|48|56|64|72|80|88|96|104|112|120|128|136|144|152|160|168|176|184|192|200|208|216|224|232|240|248|256)$/,
+    'BigInt',
+  ],
+]
+
+const typeSuggestion = typeName =>
+  TYPE_SUGGESTIONS.filter(([pattern, _]) => {
+    return typeof pattern === 'string' ? pattern === typeName : typeName.match(pattern)
+  })
+    .map(([_, suggestion]) => suggestion)
+    .find(_ => true)
```

Comment:
Is `.find(_ => true)` a way of getting the first item in the iterator? Is there a less hacky way?

Reply:
Done